filepicker/filepicker.go

```diff
@@ -0,0 +1,364 @@
+	}
+}
+
+// Init initializes the file picker model.
+func (m Model) Init() tea.Cmd {
+	return readDir(m.Path, m.ShowHidden)
+}
+
+// Update handles user interactions within the file picker model.
+func (m Model) Update(msg tea.Msg) (Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case readDirMsg:
+		m.files = msg
+	case tea.WindowSizeMsg:
+		if m.AutoHeight {
+			m.Height = msg.Height - marginBottom
+		}
+		m.max = m.Height
+	case tea.KeyMsg:
+		switch {
+		case key.Matches(msg, m.KeyMap.GoToTop):
+			m.selected = 0
+			m.min = 0
+			m.max = m.Height - 1
+		case key.Matches(msg, m.KeyMap.GoToLast):
+			m.selected = len(m.files) - 1
+			m.min = len(m.files) - m.Height
+			m.max = len(m.files) - 1
+		case key.Matches(msg, m.KeyMap.Down):
+			m.selected++
+			if m.selected >= len(m.files) {
+				m.selected = len(m.files) - 1
+			}
+			if m.selected > m.max {
+				m.min++
+				m.max++
+			}
+		case key.Matches(msg, m.KeyMap.Up):
+			m.selected--
+			if m.selected < 0 {
+				m.selected = 0
+			}
+			if m.selected < m.min {
+				m.min--
+				m.max--
+			}
+		case key.Matches(msg, m.KeyMap.PageDown):
+			m.selected += m.Height
+			if m.selected >= len(m.files) {
+				m.selected = len(m.files) - 1
+			}
+			m.min += m.Height
+			m.max += m.Height
+
+			if m.max >= len(m.files) {
+				m.max = len(m.files) - 1
+				m.min = m.max - m.Height
+			}
+		case key.Matches(msg, m.KeyMap.PageUp):
+			m.selected -= m.Height
+			if m.selected < 0 {
+				m.selected = 0
+			}
+			m.min -= m.Height
+			m.max -= m.Height
+
+			if m.min < 0 {
+				m.min = 0
+				m.max = m.min + m.Height
+			}
+		case key.Matches(msg, m.KeyMap.Back):
+			m.Path = filepath.Dir(m.Path)
+			if m.selectedStack.Length() > 0 {
+				m.selected, m.min, m.max = m.popView()
+			} else {
+				m.selected = 0
+				m.min = 0
+				m.max = m.Height - 1
+			}
+			return m, readDir(m.Path, m.ShowHidden)
+		case key.Matches(msg, m.KeyMap.Enter):
+			if len(m.files) == 0 {
+				break
+			}
+
+			f := m.files[m.selected]
+			info, err := f.Info()
+			if err != nil {
+				break
+			}
+			isSymlink := info.Mode()&os.ModeSymlink != 0
+			isDir := f.IsDir()
+
+			if isSymlink {
+				symlinkPath, _ := filepath.EvalSymlinks(filepath.Join(m.Path, f.Name()))
+				info, err := os.Stat(symlinkPath)
+				if err != nil {
+					break
+				}
+				if info.IsDir() {
+					isDir = true
+				}
+			}
+
+			if (!isDir && m.FileAllowed) || (isDir && m.DirAllowed) {
+				if key.Matches(msg, m.KeyMap.Select) {
+					selectedFile := filepath.Join(m.Path, f.Name())
+					return m, func() tea.Msg {
+						return FileSelectedMsg{selectedFile}
+					}
+				}
+			}
+
+			if !isDir {
+				break
+			}
+
+			m.Path = filepath.Join(m.Path, f.Name())
+			m.pushView()
+			m.selected = 0
+			m.min = 0
+			m.max = m.Height - 1
+			return m, readDir(m.Path, m.ShowHidden)
+		}
+	}
+	return m, nil
+}
+
+// View returns the view of the file picker.
+func (m Model) View() string {
+	if len(m.files) == 0 {
+		return "Bummer. No files found."
```

Comment:
It's probably a good idea to allow people to customize this, especially for localization. Could even be a Lip Gloss style with `SetString` applied so it could be styled accordingly.

Reply:
Ah very good point!

Reply:
Done!